README.md

```diff
@@ -1,5 +1,124 @@
+
+## Databases
+
+The CRD `PostgreSQLDatabase` specified details about a database on a specific instance.
+It is scoped as cluster wide ie. has no namespace.
+
+The main purpose of this resource is to create databases on a specific host with a specific name used by application services.
+Instances of `PostgreSQLUser` can then give a specific developer access to the database by referenceing the name.
+
+```yaml
+apiVersion: lunar.bank/v1beta1
+kind: PostgreSQLDatabase
+metadata:
+  name: user
+spec:
+  name: user
+  password:
+    value: <strong-password>
+  host:
+    value: some.host.com
+```
+
+The `password` and `host` fields are modelled like Kubernetes core's [`EnvVarSource`](https://github.com/kubernetes/api/blob/665c8a257c1af277521b08dd43d5c73570405ef0/core/v1/types.go#L1847-L1862), so it can specify raw values like above or reference ConfigMaps and Secrets.
+
+<details>
+<summary>Example of a database referencing ConfigMap and Secret resource</summary>
+
+```yaml
+apiVersion: lunar.bank/v1beta1
+kind: PostgreSQLDatabase
+metadata:
+  name: user
+spec:
+  name: user
+  password:
+    valueFrom:
+      secretKeyRef:
+        name: user-db
+        key: db.password
+  host:
+    valueFrom:
+      configMapKeyRef:
+        name: database
+        key: db.host
+```
+
+</details>
+
+The controller will ensure that a database exists on the host based on its configuration.  
+If a resources is deleted we _might_ delete the database in the future, preferrable behind a flag to avoid loosing data.
+
+## Users
+
+The CRD `PostgreSQLUser` contains metadata about the user along with its access rights to databases.
+
+The access rights are devided into read and write and specifies a `host` and `reason` as a minimum.
+This `database` field is required for writes along with `start` and `end`.
+This ensures no unnecessary capabilities are left on a user ie. after completing a support ticket.
+
+For reads, it will default to all databases on the instance and requires no time limit.
+We generally do not limit access to data but instead rely on strong audits.
+
+This is an example of a user `bso` that has read access to all databases and write access to the `user` database between 10 AM to 2 PM on september 9th.
+The read capability uses a static host name `some.host.com` and the write capability references a `database` ConfigMap on key `db.host`.
+
+```yaml
+apiVersion: lunar.bank/v1
+kind: PostgreSQLUser
+metadata:
+  name: bso
+spec:
+  name: bso
+  read:
+    - host:
+        value: some.host.com
+      reason: "I am a developer"
+  write:
+    - host:
+        valueFrom:
+          configMapKeyRef:
+            name: database
+            key: db.host
+      database: user
+      reason: "Related to support ticket LW-1234"
+      start: 2019-09-16T10:00:00Z
+      end: 2019-09-16T14:00:00Z
+```
+
+From the configuration the user will be created with an `iam_<name>` user on the host and granted rights to access the required databases.
```

Comment:
Is this the convention we are going for? Or should we use the `iam_developer_<name>` as we do in the tooling?

Reply:
That's a bug. It should be `iam_developer`. In practice I think it is easiest to make this a configurable prefix on the controller. I'll update.

```suggestion
From the configuration the user will be created with an `iam_developer_<name>` user on the host and granted rights to access the required databases.
```